packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+
+import { isFunction, isUndefined } from '@aws-amplify/ui';
+import { usePaginate } from '../hooks/usePaginate';
+import { useStore } from '../../providers/store';
+import { useAction } from '../../do-not-import-from-here/actions';
+import { LocationData, LocationItemData } from '../../actions';
+
+interface UseLocationDetailView {
+  hasNextPage: boolean;
+  hasError: boolean;
+  isLoading: boolean;
+  isPaginateNextDisabled: boolean;
+  isPaginatePreviousDisabled: boolean;
+  message: string | undefined;
+  pageItems: LocationItemData[];
+  page: number;
+  onAccessItem: (location: LocationData) => void;
+  onRefresh: () => void;
+  onPaginateNext: () => void;
+  onPaginatePrevious: () => void;
+  onAddFiles: (files: File[]) => void;
+}
+
+export type LocationDetailViewActionType =
+  | { type: 'REFRESH_DATA' } // refresh data only
+  | { type: 'RESET' } // reset view to initial state
+  | { type: 'PAGINATE_NEXT' }
+  | { type: 'PAGINATE_PREVIOUS' }
+  | { type: 'PAGINATE'; page: number }
+  | { type: 'ACCESS_ITEM'; key: string }
+  | { type: 'NAVIGATE'; location: LocationData }
+  | { type: 'ADD_FILES'; files: File[] }
+  | { type: 'SEARCH'; query: string; includeSubfolders?: boolean };
+
+interface InitialValues {
+  pageSize?: number;
+  delimiter?: string;
+}
+
+export interface UseLocationDetailViewOptions {
+  initialValues?: InitialValues;
+  onDispatch?: React.Dispatch<LocationDetailViewActionType>;
+  onActionSelect?: (type: string) => void;
+  onExit?: () => void;
+  onNavigate?: (destination: LocationData) => void;
+}
+
+const DEFAULT_PAGE_SIZE = 100;
+export const DEFAULT_LIST_OPTIONS = {
+  delimiter: '/',
+  pageSize: DEFAULT_PAGE_SIZE,
+};
+
+export function useLocationDetailView(
+  options?: UseLocationDetailViewOptions
+): UseLocationDetailView {
+  const initialValues = options?.initialValues;
+  const onActionSelect = options?.onActionSelect;
+  const onNavigate = options?.onNavigate;
```

Comment:
Nit: Just to reduce some of the boilerplate for readability
```suggestion
  const { initialValues, onActionSelect, onNavigate } = options ?? {};
```